quadra/utils/export.py

```diff
@@ -33,14 +54,52 @@ def generate_torch_inputs(
             return tuple(generate_torch_inputs(inp, device, half_precision, dtype) for inp in input_shapes)
 
         # Base case
-        inp = torch.randn((1, *input_shapes), dtype=dtype, device=device)
+        inp = torch.randn((batch_size, *input_shapes), dtype=dtype, device=device)
 
     if half_precision:
         inp = inp.half()
 
     return inp
 
 
+def extract_torch_model_inputs(
+    model: Union[nn.Module, ModelSignatureWrapper],
+    input_shapes: Optional[List[Any]] = None,
+    half_precision: bool = False,
+    batch_size: int = 1,
+) -> Optional[Tuple[Union[List[Any], Tuple[Any, ...], torch.Tensor], List[Any]]]:
```

Comment:
we can add type alias for the `inp` value for readability so in this way we can add it to docstring too.